src/user-event/type/type.ts

```diff
@@ -1,20 +1,129 @@
 import { ReactTestInstance } from 'react-test-renderer';
-import { UserEventInstance } from '../setup';
-import { dispatchHostEvent, wait } from '../utils';
+import { getConfig } from '../../config';
 import { EventBuilder } from '../event-builder';
+import { ErrorWithStack } from '../../helpers/errors';
+import { UserEventInstance } from '../setup';
+import {
+  dispatchHostEvent,
+  wait,
+  getTextRange,
+  getTextContentSize,
+} from '../utils';
+
+import { parseKeys } from './parseKeys';
+
+export interface TypeOptions {
+  skipPress?: boolean;
+  submitEditing?: boolean;
+}
 
 export async function type(
   this: UserEventInstance,
   element: ReactTestInstance,
-  text: string
-) {
-  // TODO provide real implementation
+  text: string,
+  options?: TypeOptions
+): Promise<void> {
+  if (element.type !== getConfig().hostComponentNames?.textInput) {
```

Comment:
To access host component names the function getHostComponentNames should be used instead

Reply:
Done.